schematic/schemas/data_model_json_schema.py

```diff
@@ -0,0 +1,356 @@
+import logging
+import networkx as nx
+import os
+from typing import Any, Dict, Optional, Text, List
+
+from schematic.schemas.data_model_graph import DataModelGraphExplorer
+from schematic.schemas.data_model_relationships import DataModelRelationships
+
+from schematic.utils.validate_utils import rule_in_rule_list
+logger = logging.getLogger(__name__)
+
+class DataModelJSONSchema:
+	def __init__(self, jsonld_path: str, graph:nx.MultiDiGraph,
+				 ):
+		self.jsonld_path = jsonld_path
+		self.graph = graph
+		self.DME = DataModelGraphExplorer(self.graph)
+		self.dmr = DataModelRelationships()
+		self.rel_dict = self.dmr.relationships_dictionary
+			
+	def get_array_schema(
+		self, node_range: List[str], node_name: str, blank=False
+	) -> Dict[str, Dict[str, List[str]]]:
+		"""Add a list of nodes to the "enum" key in a given JSON schema object.
+		   Allow a node to be mapped to any subset of the list
+
+		Args:
+			node_name: Name of the "main" / "head" key in the JSON schema / object.
+			node_range: List of nodes to be added to the JSON object.
+			blank: If True, add empty node to end of node list.
+				   If False, do not add empty node to end of node list.
+
+		Returns:
+			JSON object with array validation rule.
+		"""
+
+		schema_node_range_array = {
+			node_name: {
+				"type": "array",
+				"items": {"enum": node_range + [""] if blank else node_range},
+				"maxItems": len(node_range),
+			}
+		}
+
+		return schema_node_range_array
+
+	def get_non_blank_schema(
+		self, node_name: str
+	) -> Dict:  # can't define heterogenous Dict generic types
```

Comment:
You can define heterogenous dict types in a few ways such as:
`dict[Any, Any]` : literally anything for both keys and values
`dict[str, Any]` : The keys are always strings, but the value could be anything
`dict[str, Union[str, int]]` They keys are always strings, but the values vould be a string or an int

For this particular type you could use:
`dict[str, dict[str, Any]]`

But I think you might want to refer to the above comment about dataclasses.

Reply:
@linglp